source/vision/visionHandler.py

```diff
@@ -108,58 +174,50 @@ def terminateProvider(self, providerName: str) -> bool:
 			try:
 				providerInst.registerEventExtensionPoints(self.extensionPoints)
 			except Exception:
-				log.error("Error while registering to extension points for provider %s" % providerName, exc_info=True)
-		return success
+				log.error(f"Error while registering to extension points for provider {providerId}", exc_info=True)
+		if exception:
+			raise exception
 
-	def initializeProvider(self, providerName: str, temporary: bool = False) -> bool:
+	def initializeProvider(
+			self,
+			provider: providerInfo.ProviderInfo,
+			temporary: bool = False
+	) -> None:
 		"""
 		Enables and activates the supplied provider.
-		@param providerName: The name of the registered provider.
+		@param provider: The provider to initialise.
 		@param temporary: Whether the selected provider is enabled temporarily (e.g. as a fallback).
 			This defaults to C{False}.
 			If C{True}, no changes will be performed to the configuration.
-		@returns: Whether initializing the requested provider succeeded.
 		"""
-		providerCls = None
-		providerInst = self.providers.pop(providerName, None)
+		providerId = provider.providerId
+		providerInst = self.providers.pop(providerId, None)
 		if providerInst is not None:
-			providerCls = type(providerInst)
-			try:
-				providerInst.reinitialize()
-			except Exception:
-				# Purposely catch everything.
-				# A provider can raise whatever exception,
-				# therefore it is unknown what to expect.
-				log.error("Error while reinitializing provider %s" % providerName, exc_info=True)
-				return False
+			providerInst.reinitialize()

```

Comment:
I'll wrap it in a try block to make it clear. But I will re-raise the exception since this is the control flow we want anyway. Note that this is only called from places that should handle exceptions anyway.